conans/client/remote_manager_git.py

```diff
@@ -0,0 +1,167 @@
+import os
+import sys
+from distutils.dir_util import copy_tree
+from fnmatch import fnmatch
+from io import StringIO
+
+import yaml
+
+from conan.tools.files import load
+from conans.client.cmd.export import cmd_export
+from conans.errors import ConanException, PackageNotFoundException, RecipeNotFoundException
+from conans.model.recipe_ref import RecipeReference
+
+
+class GitRemoteManager:
+
+    def __init__(self, remote):
+        self._remote = remote
+        cache_folder = self._remote.url.replace("file:///", "")
```

Comment:
This does not support the host name properly https://datatracker.ietf.org/doc/html/rfc8089#section-2 😱  it can also be `file:/absolute/path` or  `file://hostname/absolute/path` as well

This protocol is super buggy with different parsers using and not using the 1,2, or 3 slashes correctly in my experience 

If you drop the third `/` that would make it a relative path? unless you are requiring 4 slashs and that i've never seen before ;p

Reply:
> This does not support the host name properly https://datatracker.ietf.org/doc/html/rfc8089#section-2 😱 it can also be file:/absolute/path or file://hostname/absolute/path as well

It is only intended to support local filesystem folders,  which means ``file:///`` is the way to specify it. It doesn't intend to support hostnames at all. In any case, this is a small implementation detail, that might be changed to any thing else, like a new Conan remote type.